app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,32 @@
+class Admin::VideosController < ApplicationController
+  before_filter :require_user
+  before_filter :require_admin
+
+  def new
+    @video = Video.new
+  end
+
+  def create
+    @video = Video.new(video_params)
+    if @video.save
+      flash[:success] = "You have successfully added the video #{@video.title}"
+      redirect_to new_admin_video_path
+    else 
+      flash[:error] = "The video could not be added. Please check the errors below."
```

Comment:
Its `flash.now` when you `render` a template. Otherwise the flash message pops on next GET request.